### ISSUE TITLE
generic FromConfig instance silently skips unsupported fields

### ISSUE THREAD

**Issue by yaitskov** [ludat/conferer#64]
I would like to see at least warnings if automatic FormConfig instance encounters field which is not supported and it is going to be ignored.

example:

```
import Katip.Core

data AppConfig {
  _appConfig_severity :: Severity;
} derive (Generic, Show)

``` 

Nonetheless Severity is Generic and Read I cannot override default value
until 
```
instance FromConfig Severity
```

Current behavior is between 2 consistent solutions:
  - fails at compile time
  - use bare Generic/Read without FromConfig
  
```
:i Severity
type Severity :: *
data Severity
  = DebugS
  | InfoS
  | NoticeS
  | WarningS
  | ErrorS
  | CriticalS
  | AlertS
  | EmergencyS
  	-- Defined in ‘Katip.Core’
instance Eq Severity -- Defined in ‘Katip.Core’
instance Ord Severity -- Defined in ‘Katip.Core’
instance Bounded Severity -- Defined in ‘Katip.Core’
instance Enum Severity -- Defined in ‘Katip.Core’
instance Generic Severity -- Defined in ‘Katip.Core’
instance Read Severity -- Defined in ‘Katip.Core’
instance Show Severity -- Defined in ‘Katip.Core’
```

**Comment by ludat** [comment]
That's right, this is because inside Conferer there is an instance for FromConfig that matches every type: [here](https://github.com/ludat/conferer/blob/7e49a39cf66c447f268930e07720873935a7f970/packages/conferer/src/Conferer/FromConfig/Internal.hs#L87) which means that all types implement FromConfig, and that that instance is not very smart (no Read or Generic instances required)

This was implemented like this mainly because of funcions: `snap`, `warp` and many other have function in their settings and I wanted to allow the programmer to override them and that meant that they needed to implement `FromConfig`, this instance is not very useful but you can override it with something more sensible using the [super override](https://conferer.ludat.io/docs/fromconfig/overrides) method.

**Comment by ludat** [comment]
but also, the fact that the Generic based instance compiles with that type is definitely a bug (we only programmed the case where the generic type is a record with only one constructor)

An easy fix could be writing the actual instance for `FromConfig Severity` which would be something like:

```haskell
import Conferer
import Conferer.FromConfig

instance FromConfig Severity where
  fromConfig = fromConfigByRead
```

that will allow you to do `./program.exe --logging.severity=DebugS`

**Comment by hughjfchen** [comment]
> but also, the fact that the Generic based instance compiles with that type is definitely a bug (we only programmed the case where the generic type is a record with only one constructor)
> 
> An easy fix could be writing the actual instance for `FromConfig Severity` which would be something like:
> 
> ```haskell
> import Conferer
> import Conferer.FromConfig
> 
> instance FromConfig Severity where
>   fromConfig = fromConfigByRead
> ```
> 
> that will allow you to do `./program.exe --logging.severity=DebugS`

Hit by this problem too. I think all user-defined sum types would have such a problem. Meanwhile, I believe `fromConfigByRead` should be `fetchFromConfigByRead`.

**Comment by ludat** [comment]
yeah, thanks for the feedback, I'll discuss this with @JuanFdS, when we decided to create the instance for `a` we feared that this could happen but expected that it wouldn't be such a common problem, again thanks for the feedback :D
